### PR TITLE
implement isomerism bug fixes plus tests

### PR DESCRIPTION
**Pull request by fgrunewald**

### CODE REVIEW

File: cgsmiles/tests/test_molecule_resolve.py [hunk 1]
```diff
@@ -179,35 +180,73 @@ def test_match_bonding_descriptors(bonds_source, bonds_target, edge, btypes):
...
+                        [(0, 1), (1, 2), (0, 3), (0, 4),
+                         (0, 5), (1, 7), (7, 6), (7, 8), (8, 9), (8, 10), (8, 11)],
+                        {}, {2: (2, 1, 6, 7, 'cis'), 7: (7, 6, 1, 2, 'cis')}),
 ))
-def test_all_atom_resolve_molecule(smile, ref_frags, elements, ref_edges):
+def test_all_atom_resolve_molecule(smile, ref_frags, elements, ref_edges, chiral, ez):
     meta_mol, molecule = MoleculeResolver.from_string(smile).resolve()
 
     # loop and compare fragments first
+    counter=0
```
Comment (pckroon) [inline cgsmiles/tests/test_molecule_resolve.py:239]:
```suggestion
    counter = 0
```

**Review by pckroon — CHANGES_REQUESTED** [review]
Do we also need do the relative accounting for rs_isomers?

File: cgsmiles/graph_utils.py [hunk 2]
```diff
@@ -63,6 +70,13 @@ def sort_nodes_by_attr(graph, sort_attr="fragid"):
         the graph to sort nodes of
     sort_attr: `abc.hashable`
         the attribute to use for sorting
+    relative_attr: tuple(str, bool)
+        a list of attributes that are sensetive
+        to the ordering of nodes (i.e. refer to
+        other nodes). The second element indicates
+        the depth. If False the value of attr are
+        node keys. If True we expect the value to
+        be an itertable with node keys.
```
[the remark this answers is not in the document]
Comment (fgrunewald) [reply cgsmiles/graph_utils.py:79]: yeah I condiered that but then wasn't worth it ^^

**Comment by fgrunewald** [comment]
@pckroon at the moment is is not required for the rs_isomer as the nodes only get annotated with '@' or '@@'. However, in the current code it is not possible to have the chiral center using the ring bond accounting. That's why it is not needed at the moment. 

The ring thing is a bit tricky because we just strip the smiles string of these characters. Doing the ring thing would actually require looking up the index belonging to a ring atom and matching it.

File: cgsmiles/graph_utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (pckroon) [reply cgsmiles/graph_utils.py:79]: The downside of doing it the "clever" way is that there will be cases it gets it wrong (deeper nesting, for example). The downside of using the bool in the argument is that it complicates the api

File: cgsmiles/graph_utils.py [hunk 3]
```diff
@@ -91,10 +91,16 @@ def sort_nodes_by_attr(graph,
...
+                iter(values)
+            except TypeError:
+                is_list = False
+            else:
+                is_list = not isinstance(values, str)
             if is_list:
                 new_values = [mapping[value] for value in values]
             else:
-               new_values = mapping[values]
+                new_values = mapping[values]
```
Comment (pckroon) [inline cgsmiles/graph_utils.py:103]: Don't forget to update the docstring :)

**Review by pckroon — CHANGES_REQUESTED** [review]
For the chiral ring things, is it possible to at least detect it and issue a warning?

**Comment by fgrunewald** [comment]
rings are fixed as long as they are within a fragment; chirality split between fragments remains a fragile thing

**Review by pckroon — CHANGES_REQUESTED** [review]
Looks good, some small nitpicks

File: cgsmiles/pysmiles_utils.py [hunk 4]
```diff
@@ -103,3 +107,57 @@ def annotate_ez_isomers(molecule):
...
+                # all hydrogen atoms are explicit in cgsmiles
+                # BUT in the input of chirality they are usual
+                # implicit. At this point we need to treat all
+                # hydrogen atoms as if they were implicit.
+                if molecule.nodes[neighbour]['element'] != 'H':
+                    neighbours.append(neighbour)
+                else:
+                    hstash = neighbour
+        if hstash:
+            neighbours.insert(1, hstash)
```
[the remark this answers is not in the document]
Comment (fgrunewald) [reply cgsmiles/pysmiles_utils.py:148]: I added an error

**Review by pckroon — APPROVED** [review]

**Pull request merged**